File: scratch/replace_products.py
import re
import os

with open('src/app/dashboard/page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

start_pattern = r'^\s*\{activeTab === "products" && \(\n\s*<div className="bg-white p-8 rounded-3xl'
end_pattern = r'</form>\n\s*</div>\n\s*\)\}\n\s*</div>\n\s*\)\}'

matches = list(re.finditer(start_pattern, content, re.MULTILINE))
if not matches:
    print('Start pattern not found!')
else:
    start_idx = matches[0].start()
    
    end_matches = list(re.finditer(end_pattern, content[start_idx:], re.MULTILINE))
    if not end_matches:
        print('End pattern not found!')
    else:
        end_idx = start_idx + end_matches[0].end()
        
        # Replacement code
        replacement = """      {activeTab === "products" && (
        <ProductManager 
          subscriptionTier={subscriptionTier}
          setIsUpgraderOpen={setIsUpgraderOpen}
          sellerProductsRaw={sellerProductsRaw}
          sellerRole={actualRole}
          isAutoApprovedUser={subscriptionTier !== "free"}
          storeName={userData?.personalName || ""}
        />
      )}"""
        
        new_content = content[:start_idx] + replacement + content[end_idx:]
        
        # Add import
        import_stmt = 'import ProductManager from "@/components/dashboard/ProductManager";\n'
        if import_stmt not in new_content:
            import_idx = new_content.find('import OrderManager')
            if import_idx != -1:
                new_content = new_content[:import_idx] + import_stmt + new_content[import_idx:]
        
        with open('src/app/dashboard/page.tsx', 'w', encoding='utf-8') as f:
            f.write(new_content)
        print('Successfully replaced products tab and added import.')

File: scratch/replace_products.js
const fs = require('fs');

const filePath = 'src/app/dashboard/page.tsx';
let content = fs.readFileSync(filePath, 'utf-8');

const startPattern = /^\s*\{activeTab === "products" && \(\n\s*<div className="bg-white p-8 rounded-3xl/m;
const endPattern = /<\/form>\n\s*<\/div>\n\s*\)\}\n\s*<\/div>\n\s*\)\}/;

const startMatch = content.match(startPattern);
if (!startMatch) {
    console.log('Start pattern not found!');
    process.exit(1);
}

const startIdx = startMatch.index;
const contentFromStart = content.slice(startIdx);
const endMatch = contentFromStart.match(endPattern);

if (!endMatch) {
    console.log('End pattern not found!');
    process.exit(1);
}

const endIdx = startIdx + endMatch.index + endMatch[0].length;

const replacement = `      {activeTab === "products" && (
        <ProductManager 
          subscriptionTier={subscriptionTier}
          setIsUpgraderOpen={setIsUpgraderOpen}
          sellerProductsRaw={sellerProductsRaw}
          sellerRole={actualRole}
          isAutoApprovedUser={subscriptionTier !== "free"}
          storeName={userData?.personalName || ""}
        />
      )}`;

let newContent = content.slice(0, startIdx) + replacement + content.slice(endIdx);

const importStmt = 'import ProductManager from "@/components/dashboard/ProductManager";\n';
if (!newContent.includes(importStmt)) {
    const importIdx = newContent.indexOf('import OrderManager');
    if (importIdx !== -1) {
        newContent = newContent.slice(0, importIdx) + importStmt + newContent.slice(importIdx);
    }
}

fs.writeFileSync(filePath, newContent, 'utf-8');
console.log('Successfully replaced products tab and added import.');
